File: 217 - Contains Duplicate/solution.js
/**
 * @param {number[]} nums
 * @return {boolean}
 * 
 * Time Complexity - O(n)
 * Space Complexity - O(n)
 */
var containsDuplicate = function(nums) {
                
    const mset = new Set();
    
    for(num of nums){
        if(mset.has(num)){
            return true;
        }
        mset.add(num);
    }

    return false;
};


/**
 * @param {number[]} nums
 * @return {boolean}
 * 
 * Time Complexity - O(n)
 * Space Complexity - O(n)
 */
var containsDuplicate = function(nums) {
                
    const mset = new Set(nums);
    
    return mset.size == nums.length ? false : true;
};

File: 217 - Contains Duplicate/solution.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        
        // --------------------- Using Sorting ---------------------
        // Arrays.sort(nums);

        // for(int i=1; i<nums.length; i++){
        //     if(nums[i] == nums[i-1]){
        //         return true;
        //     }
        // }

        // return false;

        // --------------------- Using Hashset ---------------------
        HashSet<Integer> set = new HashSet<>();

        for(int num: nums){
            if(set.contains(num)){
                return true;
            }
            set.add(num);
        }

        return false;
    }
}